code/modules/admin/verbs/adminpm.dm

```diff
@@ -46,6 +46,25 @@
 		else		src << "<font color='red'>Error: Private-Message: Client not found. They may have lost connection, so try using an adminhelp!</font>"
 		return
 
+	if (src.handle_spam_prevention(msg,MUTE_ADMINHELP))
+		return
+
+	var/recieve_pm_type = "Player"
+	if(holder)
+		//mod PMs are maroon
+		//PMs sent from admins and mods display their rank
+		if(holder)
+			if(!C.holder && holder && holder.fakekey)
+				recieve_pm_type = "Admin"
+			else
+				recieve_pm_type = holder.rank
+
+	else if(C && !C.holder)
+		src << "<span class='warning'>Error: Admin-PM: Non-admin to non-admin PM communication is forbidden.</span>"
+		return
+
+	msg = sanitize(msg)
```

Comment:
```dm
if (!check_rights(R_SERVER|R_DEBUG, 0))
	msg = sanitize(msg)
```

Having admins be able to HTML inject for top keks is a part of the Aurora experience, I'm afraid.

Reply:
It was the party poopers who removed it from Bay not long ago. 😞